bluetooth: host: Add CS unit tests and helper API

olivier-le-sage opened the pull request:
Add CS unit tests and helper API

olivier-le-sage committed:
bluetooth: host: Add helper function for parsing PCTs

The 12-bit signed values for the results of PBR are a bit
cumbersome, so this adds a helper function to make it easier to work
with the HCI formatted steps.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage commented on subsys/bluetooth/host/cs.c:
```diff
@@ -240,6 +240,24 @@ static void reset_reassembly_results(void)
+/** @brief Converts PCT to a pair of int16_t
+ *
+ */
+struct bt_le_cs_iq_sample bt_le_cs_parse_pct(const uint8_t pct[3])
```
@alwa-nordic fyki

theob-pro commented on tests/bluetooth/host/cs/bt_le_cs_parse_pct/CMakeLists.txt:
```diff
@@ -0,0 +1,25 @@
+
+find_package(Zephyr COMPONENTS unittest REQUIRED HINTS $ENV{ZEPHYR_BASE})
+
+project(bt_step_data_parse)
```
The project should be `bt_parse_pct`, no?

theob-pro commented on tests/bluetooth/host/cs/mocks/CMakeLists.txt:
```diff
@@ -0,0 +1,15 @@
+add_library(mocks STATIC
```
Is it fine to not have license identifier there?

olivier-le-sage replied:
added

theob-pro commented on tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h:
```diff
@@ -0,0 +1,10 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Dummy header file to avoid compiler errors
+ * intentionally left blank
+ */
```
Which compiler error? Is it something that should be reported and fixed later?

olivier-le-sage replied:
I copied these over from a different host unit test suite. I'm not sure if they're all really needed, let me check.

olivier-le-sage replied:
seems not :)

theob-pro commented on tests/bluetooth/host/cs/mocks/zephyr/linker/linker-defs.h:
```diff
@@ -0,0 +1,17 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Undefine definitons that conflict in the unit_testing board build. */
+
+#ifdef DT_NODE_HAS_STATUS
+#undef DT_NODE_HAS_STATUS
+#endif
+
+#ifdef DT_FOREACH_OKAY_HELPER
+#undef DT_FOREACH_OKAY_HELPER
+#endif
+
+#define DT_FOREACH_OKAY_HELPER(x)
```
Also wondering about that. Should it be addressed by whoever maintain the unit_testing board?

olivier-le-sage committed:
bluetooth: host: Unit tests for bt_le_cs_step_data_parse

Add unit tests for bt_le_cs_step_data_parse.
Also exit early (without calling the function pointer) if the next
step would seem to read out of bounds.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>
